### DISCUSSION TITLE
iopool recuperation de l'id du bassin

### DISCUSSION THREAD

**Discussion by Chrisdie59** [mguyard/appdaemon-iopoolpumpmanager#33] [Q&A]
Bonsoir

je n'arrive pas a recuperer l'id de mon bassin sur Windows
quelle appilication dois utiliser ?
merci de votre aide

**Comment by mguyard** [comment]
Bonjour,

Quelle erreur avez-vous avec les commandes fournient dans la documentation ?
Quelle version de Windows utilisez-vous ?

**Comment by Chrisdie59** [comment]
Bonjour voici l'erreur que j'ai sous windows 11
Invoke-WebRequest : Le nom distant n'a pas pu être résolu: 'api.iopool.com'
Au caractère Ligne:3 : 13
+ $response = Invoke-WebRequest -Uri 'https://api.iopool.com/v1/pools/' ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExcepti 
   on
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

**Reply by Chrisdie59** [reply]
Merci de votre aide

**Reply by mguyard** [reply]
Pouvez-vous essayer cette commande et me partager la réponse reçue
```
nslookup api.iopool.com
```

**Reply by mguyard** [reply]
C'est surprenant que le nslookup retourne bien les adresses IP alors que dans la commande ci-dessus il indiquait ne pas pouvoir résoudre :

```
Invoke-WebRequest : Le nom distant n'a pas pu être résolu: 'api.iopool.com'
```

Est-ce que vous avez bien changé la clé API pour la remplacer par la votre dans la commande ?
Est-ce que vous pouvez ajouter la commande suivante à la suite des commandes indiquées dans la doc et me fournir l'ensemble du resultat ? 
```
$statusCode = $response.StatusCode
```

**Reply by Chrisdie59** [reply]
je met bien ma cle API recuperer sur mon application
voici une copie du code avec l'ajout demandé et la réponse recue 
![Capture d’écran 2024-08-18 140502](https://github.com/user-attachments/assets/15342d88-134b-466a-9bdb-471c600cac97)

**Reply by mguyard** [reply]
Je suis plus habitué à Linux/mac mais pour moi il faut que chaque ligne commençant par un $ soit envoyé une à une.
Dans la capture il semble que tout soit à la suite.
Faite chaque ligne l'une après l'autre

**Reply by Chrisdie59** [reply]
j'ai tous remis a la ligne mais j'ai la meme erreur

Invoke-WebRequest : Le nom distant n'a pas pu être résolu: 'api.iopool.com'
Au caractère Ligne:3 : 13
+ $response = Invoke-WebRequest -Uri 'https://api.iopool.com/v1/pools/' ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebException
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand
 ualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

**Reply by mguyard** [reply]
Bizarre. Comme vous êtes sur Windows 11, vous devriez pouvoir utiliser la commande Linux avec CURL qui est dans la doc.
Si la commande ne fonctionne pas, remplacez dans la commande curl, curl par curl.exe

**Comment by Chrisdie59** [comment]
Bonsoir
Voila le retour de la requete

PS C:\WINDOWS\system32> PS C:\WINDOWS\system32> nslookup api.iopool.com
nslookup : R‚ponse ne faisant pas autorit‚ :
Au caractère Ligne:1 : 1
+ nslookup api.iopool.com
+ ~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (R‚ponse ne faisant pas autorit‚ ::String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Serveur :   UnKnown
Address:  192.168.1.254

Nom :    d32jcdry765fns.cloudfront.net
Addresses:  52.84.174.88
	  52.84.174.94
	  52.84.174.38
	  52.84.174.58
Aliases:  api.iopool.com
